src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -290,15 +303,16 @@ sealed private[immutable] trait LazyListOps[+A, +CC[+X] <: LinearSeq[X] with Laz
 
   override def filterNot(pred: A => Boolean): C = filterImpl(pred, isFlipped = true)
 
-  private[immutable] def filterImpl(p: A => Boolean, isFlipped: Boolean): C = {
-    // optimization: drop leading prefix of elems for which f returns false
-    // var rest = this dropWhile (!p(_)) - forget DRY principle - GC can't collect otherwise
-    var rest: CC[A] = coll
-    while (rest.nonEmpty && p(rest.head) == isFlipped) rest = rest.tail
-    // private utility func to avoid `this` on stack (would be needed for the lazy arg)
-    (if (rest.nonEmpty) iterableFactory.filteredTail(rest, p, isFlipped)
-    else iterableFactory.empty).asInstanceOf[C]
-  }
+  private[immutable] def filterImpl(p: A => Boolean, isFlipped: Boolean): C =
+    suspend0 {
```

Comment:
just using `suspend` here complains that we got `C` but expected `CC[A]`

Reply:
You can add a constraint that `C <: CC[A]`, if that helps. But such constraints usually make things hard to abstract over.

Reply:
`C` is defined as
```scala
+C <: CC[A] with LazyListOps[A, CC, C]
```

The problem (I think) is that we don't know that the `iterableFactory` is going to return the _exact same collection type_ (`C`) when asked to produce a collection type wrapping `A`, even though we know/hope that `C =:= CC[A]`.